packages/babel-template/test/index.js

```diff
@@ -133,7 +133,13 @@ describe("babel-template", function() {
           ID;
           ANOTHER_ID;
         `)({ ID: t.identifier("someIdent") });
-      }).to.throw(Error, 'No substitution given for "ANOTHER_ID"');
+      }).to.throw(
+        Error,
+        `Error: No substitution given for "ANOTHER_ID". If this is not meant to be a
```

Comment:
ES6 template literals take raw string, so if you have multi-line string, then it will automatically add new Line character there. However, when you use + (with strings), you are actually doing concatenation and not putting any new Line character there. I would suggest you to convert both of them into template literals, that would automatically add new line character in both. Hence resolving test!

Reply:
thanks @nveenjain 
I forgot to include it [in my comment](https://github.com/babel/babel/pull/7255#issuecomment-359852487), but locally both implementations are seemingly matching to me in terms of implementation, yet still getting a test failure on string construction.  (I've since updated my comment to show everything)

I can push my changes if it helps.

Reply:
this one is still constructed with multiline template string, you just have to adjust this one so it is constructed like the other one (the other one is correct)

Reply:
@Andarist 
Do you mean in this PR?  Yes I understand they are different and I know why they are incompatible, but if you see the updated link from [my comment above](https://github.com/babel/babel/pull/7255#discussion_r163566474), locally you will see I have them identical.  Still struggling with the difference between this
```
        const placeholderName = placeholder.name;

        throw new Error(
          `Error: No substitution given for "${placeholderName}". If this is not meant to be a 
            placeholder you may want to consider passing one of the following options to babel-template:
            - { placeholderPattern: false, placeholderWhitelist: new Set(['${placeholderName}'])}
            - { placeholderPattern: /^${placeholderName}$/ }`,
        );
```

and this

```
      }).to.throw(
        Error,
        `Error: No substitution given for "ANOTHER_ID". If this is not meant to be a
          placeholder you may want to consider passing one of the following options to babel-template:
          - { placeholderPattern: false, placeholderWhitelist: new Set(['ANOTHER_ID']) }
          - { placeholderPattern: /^ANOTHER_ID$/ }`,
      );
```

🤒

Reply:
notice that in the source file (when throwing) the string is created (in the PR) with concatenation (`+`) and in tests it's defined with multiline template literal

This:
```js
		const str1 = `first line` +
												`second line`
```
and this:
```js
		const str2 = `first line
												second line`
```
are not equivalents.
```js
str1 === 'first linesecond line'
str2 === first line\n\t\t\t\t\t\t\t\t\t\t\t\tsecond line
```

Reply:
@Andarist 
Understood that is what is causing the failure in this PR specifically, but my comment was just trying to demonstrate what I was trying locally to fix the failure here but also avoiding pushing intentionally failing commits since I'm not getting it work in the way my comment illustrates.

I suppose that keeping my local work local is causing more confusion, so I will just push my latest changes.

Appreciate you helping out, sorry for any confusion, hopefully the issue I'm seeing will be better highlighted in a few moments.

Reply:
> Error: No substitution given for "${placeholderName}". If this is not meant to be a 

@thescientist13 the difference is an extra space after "a" in your input vs output.

Reply:
Thanks @existentialism !  I think part of the issue is that what I see in Travis is different output than what I see locally.  Will need to see if I need to align my local environment
- OSX: 10.13.2
- node: 6.9.1
- npm: 5.6.0

Reply:
@thescientist13 i pushed a change to your branch that should fix it

Reply:
wow, thanks @existentialism !!

I'll certainly never underestimate a string again... 😌